src/v/cluster/feature_manager.cc

```diff
@@ -288,4 +288,61 @@ ss::future<> feature_manager::do_maybe_update_active_version() {
     vlog(clusterlog.info, "Updated cluster version to {}", max_version);
 }
 
+ss::future<std::error_code>
+feature_manager::write_action(cluster::feature_update_action action) {
+    // TODO: validate earlier that the feature name is known to us
+
+    // Validate that teh feature is in a state compatible with the
```

Comment:
is that validation duplicated in `feature_table` ? If we need a strict validation here we should use linearizable barrier and replicate with expected term the same way as it is done in `persisted_stm`

Reply:
This isn't really meant to be strict (it's safe to bad transitions to the log because on replay they'll just be dropped), but maybe it's neater if I make it that way.  :thinking: